File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    for g in s:
        if g == '(':
            stack.append(g)
        else:
            if len(stack) > 0 and stack[-1] == '(':
                stack.pop()
            else:
                stack.append(g)
    
    if len(stack) == 0:
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        for(int i=0;i<s.length();i++){
            Character str = s.charAt(i);
            if (stack.size() == 0) {
                stack.push(str);
            } else {
                if (str == '(') {
                    stack.push(str);
                } else {
                    Character top = stack.peek();
                    if (top == '(') {
                        stack.pop();
                    } else {
                        return false;
                    }
                }
            }
        }
                    
        if (stack.size() > 0) {
            answer = false;
        }
        
        return answer;
    }
}